nf_core/pipelines/containers_utils.py

```diff
@@ -0,0 +1,132 @@
+            )
+            return out_str
+
+        except RuntimeError as e:
+            log.error("Running 'nextflow inspect' failed with the following error:")
+            raise UserWarning(e)
+
+    def generate_all_container_configs(self, default_config: str) -> None:
+        """Generate the container configuration files for all platforms."""
+        containers: dict[str, dict[str, str]] = {
+            "docker_amd64": {},
+            "docker_arm64": {},
+            "singularity_oras_amd64": {},
+            "singularity_oras_arm64": {},
+            "singularity_https_amd64": {},
+            "singularity_https_arm64": {},
+            "conda_amd64_lockfile": {},
+            "conda_arm64_lockfile": {},
+        }
+        for line in default_config.split("\n"):
+            if line.startswith("process"):
+                pattern = r"process { withName: \'(.*)\' { container = \'(.*)\' } }"
+                match = re.search(pattern, line)
+                if match:
+                    try:
+                        module_name = match.group(1)
+                        container = match.group(2)
+                    except AttributeError:
+                        log.warning(f"Could not parse container for process {line}")
+                        continue
+                else:
+                    continue
+                containers["docker_amd64"][module_name] = container
+        for module_name in containers["docker_amd64"].keys():
+            # Find module containers in meta.yml
+            if "_" in module_name:
+                module_path = Path(module_name.split("_")[0].lower()) / module_name.split("_")[1].lower()
+            else:
+                module_path = Path(module_name.lower())
+
+            try:
+                with open(self.workflow_directory / "modules" / self.org / module_path / "meta.yml") as fh:
+                    meta = yaml.safe_load(fh)
+            except FileNotFoundError:
+                log.warning(f"Could not find meta.yml for {module_name}")
+                continue
+
+            platforms: dict[str, list[str]] = {
+                "docker_arm64": ["docker", "linux_arm64", "name"],
+                "singularity_oras_amd64": ["singularity", "linux_amd64", "name"],
+                "singularity_oras_arm64": ["singularity", "linux_arm64", "name"],
+                "singularity_https_amd64": ["singularity", "linux_amd64", "https"],
+                "singularity_https_arm64": ["singularity", "linux_arm64", "https"],
+                "conda_amd64_lockfile": ["conda", "linux_amd64", "lock_file"],
+                "conda_arm64_lockfile": ["conda", "linux_arm64", "lock_file"],
+            }
+
+            for p_name, p_keys in platforms.items():
+                try:
+                    containers[p_name][module_name] = meta["containers"][p_keys[0]][p_keys[1]][p_keys[2]]
```

Comment:
```suggestion
            for p_name, (runtime, arch, protocol) in platforms.items():
			    try:
			        containers[p_name][module_name] = meta["containers"][runtime][arch][protocol]
```
makes it a bit more readable